stable/prometheus-operator/templates/exporters/core-dns/servicemonitor.yaml

```diff
@@ -3,7 +3,7 @@ apiVersion: monitoring.coreos.com/v1
 kind: ServiceMonitor
 metadata:
   name: {{ template "prometheus-operator.fullname" . }}-coredns
-  namespace: {{ template "prometheus-operator.namespace" . }}
+  namespace: "kube-system"
```

Comment:
Suggest that this is only changed to kube-system if the namespace selectors are ignored.

Reply:
Should I also only modify the namespaces for `kube-api-server`, `kube-controller-manager`, and `kubelet` ServiceMonitors if `ignoreNamespaceSelectors` is selected or does this only apply to coredns?

My initial reasoning for the namespaces directly was to keep the namespace consistent regardless of the option selected. That way, redeploying the helm chart with `ignoreNamespaceSelectors` doesn't have the unintentional side-effect of moving the ServiceMonitors outside the original deployment namespace.

Reply:
Wouldn't turning this on make these monitors fail to work?

Reply:
Yes, if the user turns on `ignoreNamespaceSelectors`, we'll have to modify the namespaces for all four components to be located in the namespace their target resides in (as this PR currently does). Otherwise the monitors will fail to work.

However, my question is whether these namespaces should only be modified only if `ignoreNamespaceSelectors` is selected for all 4 components (which is what your initial comment seems to suggest) i.e.

```
{{- if .Values.prometheusSpec.ignoreNamespaceSelectors }}
    namespace: "kube-system"
{{- else }}
    namespace: {{ template "prometheus-operator.namespace" . }}
{{- end }}
```

Or if it should be modified in general within Prometheus Operator regardless of whether `.Values.prometheusSpec.ignoreNamespaceSelectors` is enabled to keep the namespace of the resources consistent regardless of the option selected (i.e. the current format of the PR).